mavlink_vehicles/mavlink_vehicles.cc

```diff
@@ -467,18 +466,18 @@ void msghandler::handle(mav_vehicle &mav, const mavlink_message_t *msg)
 
 mav_vehicle::mav_vehicle(int socket_fd)
 {
-    // Store socket
-    this->sock = socket_fd;
+    if (!check_socket(socket_fd)) {
+        print_verbose(
+            "Error: the socket provided to the constructor is invalid\n");
+        return;
+    }
 
     // The system_id must be unique for each instance of the mav_vehicle. We
     // will use the port associated to the socket as the system_id of this
-    // instance.
+    // instance if system_id is not provided by the caller.
     struct sockaddr_storage our_addr = {0};
     socklen_t our_addr_len = sizeof(our_addr);
-    if (getsockname(this->sock, (struct sockaddr *)&our_addr, &our_addr_len) == -1) {
-        print_verbose("The socket provided to the constructor is invalid\n");
-        this->system_id = defaults::system_id;
-    }
+    getsockname(this->sock, (struct sockaddr *)&our_addr, &our_addr_len);
```

Comment:
You are using this->sock before attribution here. Also, this doesn't look good since it may lead to a TOCTOU bug. I'm not sure what is the best solution here. Maybe you could change check_socket() function to something else that provides you with the info you need and still allow for easy validation of the socket.